auth/presentation/src/main/java/com/sergius/auth/presentation/login/LoginViewModel.kt

```diff
@@ -34,15 +35,17 @@ class LoginViewModel(
                 snapshotFlow { _state.value.passwordState.password.text }
             ) { email, password ->
                 val isEmailValid = userDataValidator.isValidEmail(email = email.toString())
-                _state.value = _state.value.copy(
-                    emailState = _state.value.emailState.copy(isEmailValid = isEmailValid),
-                    canLogin = isEmailValid && password.isNotEmpty()
-                )
+                _state.update {
+                    it.copy(
+                        emailState = _state.value.emailState.copy(isEmailValid = isEmailValid),
```

Comment:
Here a race condition could technically still happen. That's why `update` provides you the captured state as a lambda parameter, so you could replace this with `it.emailState.copy()` to prevent this. Otherwise you're reading the state value outside the captured value in the `update` block